pkg/common/url.go

```diff
@@ -107,3 +111,69 @@ func NormalizeURLPath(p string) string {
 
 	return string(res)
 }
+
+// Sends an http request
+// ignore expectedStatusCode by setting it to 0
+func SendHTTPRequest(method string,
+	requestURL string,
+	body []byte,
+	headers map[string]string,
+	cookies []*http.Cookie,
+	expectedStatusCode int,
+	insecure bool) ([]byte, int, error) {
+
+	var client *http.Client
+
+	// create client (secure or not)
+	if insecure {
```

Comment:
you can make it more concise by

```
client := &http.Client{
  Timeout: timeout,
}
if (insecure) {
  client.Transport = &http.Transport{
    TLSClientConfig: &tls.Config{
      InsecureSkipVerify: true,
    },
  }
}
```

Note that I've added `timeout` to the client to avoid requests waiting forever for a reply